# nim-stuff

Exploring NIM for bioinformatics:
 * [Dereplicate sequences](https://github.com/telatin/nim-stuff/tree/master/derep#readme) - dereplicate sequences with support for USEARCH style "size=INT;" annotations that will be taken into account.
 
:book: My notes on **[Nim for Bioinformatics](https://andrea-telatin.gitbook.io/nim-bioinformatics/)**

## Explorations
 * [threads example](https://github.com/telatin/nim-stuff/tree/master/threads#readme)
 * [FASTX dereplication](https://github.com/telatin/nim-stuff/tree/master/projects/dereplication#readme)
 
## External resources:

#### Bioinformatics
 * [hts-nim-tools](https://github.com/brentp/hts-nim-tools), for BAM/CRAM processing (Brent Pedersen)
   * Brent is a pioneer in nim for bioinformatics so it's definitely worth [checking his GitHub page](https://github.com/search?q=user%3Abrentp+nim)
 * [biofast](https://github.com/lh3/biofast/), for BED and FASTA parsing (Heng Li) 
 * [sequences (nimbioseq)](https://github.com/jhbadger/nimbioseq)

#### Data analysis
 * [NimData (dataframes)](https://github.com/bluenote10/NimData)
 
#### Option parsing
 * [parseopt](https://nim-lang.org/docs/parseopt.html) - the vanilla parser in the standard library
 * [argparse](https://www.iffycan.com/nim-argparse/argparse.html) - `argparse`-style parser, probably not super mature
 * [docopt](https://github.com/docopt/docopt.nim#readme) - [docopt](https://www.docopt.org)-style parser, I find it useful

#### Other
 * [stats](https://github.com/treeform/statistics) - statistics (via @lf-araujo)
